### PR TITLE
telegram-desktop: make libXcursor hack optional

### PR DESCRIPTION
**Pull request by 9ary**
## Description of changes

I use nixos stable, but I pull tdesktop from unstable. Lately, I've been unable to open links. The following output shows up when running in a terminal:

```
XPCOMGlueLoad error for file /nix/store/25rzm9fz1qljrws497xqcwl9pjd3nq6d-firefox-nightly-bin-unwrapped-121.0a1/lib/firefox-bin-121.0a1/libmozgtk.so:
/nix/store/whypqfa83z4bsn43n4byvmw80n4mg3r8-glibc-2.37-45/lib/libc.so.6: version `GLIBC_2.38' not found (required by /nix/store/ncx0rw8c2b8bvbxfqnsvs70vqw35a28k-libXcursor-1.2.1/lib/libXcursor.so.1)
Couldn't load XPCOM.
```

Clearly it's trying to load a newer libXcursor than it should, and fails because that's built against a newer version of glibc.
I've tracked this down to  #196269.

To keep everyone happy, and avoid unnecessary rebuilds or carrying custom wrappers outside of nixpkgs, this PR splits the telegram-desktop derivation into a new -unwrapped package, and the wrapper on its own, as is common for various other packages.
It then adds an extra parameter to the package to control the inclusion of the xcursor hack (on by default).

For what it's worth, on my setup (sway), I don't have any problems with the cursor theme, but regardless being unable to open links is a more breaking issue than the cursor theme not applying.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/networking/instant-messengers/telegram/telegram-desktop/wrapper.nix [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    wrapperArgs=(
+      "''${gappsWrapperArgs[@]}"
+      "''${qtWrapperArgs[@]}"
+      --suffix PATH : ${lib.makeBinPath [ xdg-utils ]}
+    )
+    if ${lib.boolToString xcursorHack}; then
+      # Hack to load libXcursor on some setups
+      # https://github.com/NixOS/nixpkgs/issues/181176
+      wrapperArgs+=(--prefix LD_LIBRARY_PATH : "${xorg.libXcursor}/lib")
+    fi
```
Comment (chayleaf) [inline pkgs/applications/networking/instant-messengers/telegram/telegram-desktop/wrapper.nix:41]:
```suggestion
      ${lib.optionalString xcursorHack ''    
        # Hack to load libXcursor on some setups
        # https://github.com/NixOS/nixpkgs/issues/181176
        --prefix LD_LIBRARY_PATH : "${xorg.libXcursor}/lib"
      ''}
    )
```

Comment (9ary) [reply pkgs/applications/networking/instant-messengers/telegram/telegram-desktop/wrapper.nix:41]: Yeah this is nicer, thanks!

**Comment by chayleaf** [comment]
Result of `nixpkgs-review pr 265894` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>telegram-desktop</li>
    <li>telegram-desktop-unwrapped</li>
  </ul>
</details>

**Review by chayleaf — APPROVED** [review]

**Comment by NickCao** [comment]
Would you like to also take a look at https://github.com/NixOS/nixpkgs/issues/181176, since the hack seems to not work consistently, we might just drop it.

**Comment by 9ary** [comment]
I have low motivation to debug this and I wouldn't even know where to start since it doesn't repro on my machine, but I can take a look.

**Comment by NickCao** [comment]
Another option would to backport the telegram-desktop updates to stable, AFAIK mixing (graphical) software from different channels creates a lot of issues (from input methods, to opengl drivers) and what you've experiences is just an example from them.

**Comment by 9ary** [comment]
Yeah, I'm aware. I feel like backports would be justifiable for telegram because of the server-side breaking changes.

**Comment by 9ary** [comment]
As per https://github.com/NixOS/nixpkgs/issues/181176#issuecomment-1797108019, let's drop this and just revert the hack entirely.

**Pull request closed without merging**